arangod/RocksDBEngine/RocksDBKey.cpp

```diff
@@ -149,6 +150,36 @@ void RocksDBKey::constructMdiIndexValue(uint64_t indexId,
   TRI_ASSERT(_buffer->size() == keyLength);
 }
 
+void RocksDBKey::constructVectorIndexValue(uint64_t indexId,
+                                           std::uint64_t listNumber) {
+  _type = RocksDBEntryType::VectorVPackIndexValue;
+  // Key contains: indexId(uint64_t) + listNumber(uint64_t)
+  size_t keyLength = sizeof(uint64_t) + sizeof(std::uint64_t);
```

Comment:
Is there a reason for not using `std::uint64_t` for both arguments? Just curious.

Reply:
Yes, Max pointed out in previous comments that using std::size_t is not good since we might get compiled on 32-bit machines. Then, the `size_t` would change, and the data in RocksDB would not be compatible between architectures. This discussion https://github.com/arangodb/arangodb/pull/21184#discussion_r1830876498